src/pgen/progenitor.cpp

```diff
@@ -0,0 +1,178 @@
+// © 2021. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+
+
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "geometry/geometry.hpp"
+#include "monopole_gr/monopole_gr.hpp"
+#include "microphysics/eos_phoebus/eos_phoebus.hpp"
+#include "pgen/pgen.hpp"
+#include "phoebus_utils/ascii_reader.hpp"
+#include "progenitor/progenitordata.hpp"
+
+// Homologously collapsing star.
+namespace homologous {
```

Comment:
So in most places, namespaces should be capitalized but because of insane macro magic, `pgen` is special. The namespace needs to be exactly the name you want for the problem when requesting it in the input deck.

Reply:
i.e., this one is fine. :)